src/librustc_mir/hair/pattern/mod.rs

```diff
@@ -714,6 +715,7 @@ impl<'a, 'tcx> PatternContext<'a, 'tcx> {
                 },
                 user_ty: PatternTypeProjection::from_user_type(user_ty),
                 user_ty_span: span,
+                variance: ty::Variance::Invariant,
```

Comment:
Here we should use `Covariant` I believe

Reply:
Fixed.